ref/general/application-hardening.adoc

```diff
@@ -0,0 +1,108 @@
+However, because the web container serves files from the WAR root, it must never contain any content that users shouldn't see, including property or class files.
+If you must place such information in the WAR file, place it in the `WEB-INF` directory, as permitted by the servlet specification.
+Information in this directory is never served by the web container.
+
+== Secure servlets
+Servlets can be served by class name or with a normal URL alias.
+By default, Open Liberty does not allow you to serve servlets by class name.
+Instead of defining a mapping for each servlet, a single generic URL (such as `/servlet`) serves all servlets.
+The component of the path after the base is assumed to be the class name for the servlet.
+For example, `/servlet/com.ibm.sample.MyServlet` refers to the `com.ibm.sample.MyServlet` servlet class.
+
+You can serve servlets by class name by setting the `serveServletsByClassnameEnabled` property to `true` in the `ibm-web-ext.xmi` file.
+For production purposes, don't enable this property.
+Enabling it makes it possible for anyone that knows the class name of any servlet to directly invoke it.
+
+=== Servlet aliases
+Servlets are secured by URLs.
+Each URL that is secured must be specified in the `web.xml` file.
+Servlets can have more than one alias if multiple URLs access the same servlet class.
+If a servlet has more than one alias or there are many servlets, it can be easy to forget to secure an alias.
+Because Open Liberty secures URLs and not the underlying classes, if just one servlet URL is insecure, then an intruder might be able to bypass security.
+To alleviate this problem, use wildcards to secure servlets when possible.
+If using wildcards isn't an option, carefully double check your `web.xml` file before deployment to ensure that all servlet aliases are secure.
+
+When you specify authorization constraints for servlets, you must either list no methods or list all methods for a servlet, including `GET`, `POST`, `PUT`, `HEAD`, etc.
+In the following secure example, endpoints in the web application are unreachable by default.
+If a new servlet is added, you must add a security constraint for the new URL so that anyone can use it:
+
+// Is it okay to use this example? Did it come from the other article?
+[source,xml]
+----
+<security-constraint>
+  <web-resource-collection>
+    <web-resource-name>DefaultDeny</web-resource-name>
+    <url-pattern>/*</url-pattern>
+  </web-resource-collection>
+  <auth-constraint>
+    <role-name>NoAccess</role-name>
+  </auth-constraint>
+</security-constraint>
+----
+
+== Error handler
+When errors occur in a web application, an error message is displayed to the user.
+By default, Open Liberty displays a raw exception stack dump of the error.
+Not only is this output unfriendly to users, but it also reveals information about the application, including names of classes and methods.
```

Comment:
I think maybe change "including names of classes and methods" to "that includes names of classes and methods" to use a more active voice.

Reply:
Updated